extensions/common/vault/vault-hashicorp/src/main/java/org/eclipse/edc/vault/hashicorp/HashicorpVaultExtension.java

```diff
@@ -59,15 +69,54 @@ public PrivateKeyResolver privateKeyResolver() {
         return privateKeyResolver;
     }
 
+    @Provider
+    public HashicorpVaultClient hashicorpVaultClient() {
+        return hashicorpVaultClient;
+    }
+
     @Override
     public void initialize(ServiceExtensionContext context) {
-        var config = HashicorpVaultClientConfig.create(context);
-        var client = new HashicorpVaultClient(config, httpClient, typeManager.getMapper());
-
-        vault = new HashicorpVault(client, context.getMonitor());
+        scheduledExecutorService = executorInstrumentation.instrument(Executors.newSingleThreadScheduledExecutor(), NAME);
+        var monitor = context.getMonitor().withPrefix(NAME);
+        var config = HashicorpVaultConfig.create(context);
+        initHashicorpVaultClient(config, monitor);
+        vault = new HashicorpVault(hashicorpVaultClient, monitor);
         privateKeyResolver = new VaultPrivateKeyResolver(vault);
+        context.registerService(CertificateResolver.class, new HashicorpCertificateResolver(vault, monitor));
+    }
 
-        context.registerService(CertificateResolver.class, new HashicorpCertificateResolver(vault, context.getMonitor()));
+    @Override
+    public void start() {
+        if (isTokenRenewable) {
+            var tokenRenewResult = hashicorpVaultClient.renewToken();
+
+            if (tokenRenewResult.failed()) {
+                throw new EdcException("[%s] Initial token renewal failed: %s".formatted(NAME, tokenRenewResult.getFailureDetail()));
+            }
+
+            var token = tokenRenewResult.getContent();
+            hashicorpVaultClient.scheduleNextTokenRenewal(token.getTimeToLive());
+        }
+    }
+
+    @Override
+    public void shutdown() {
+        scheduledExecutorService.shutdownNow();
     }
 
+    private void initHashicorpVaultClient(HashicorpVaultConfig config, Monitor monitor) {
```

Comment:
nit: i don't like void-returning init methods too much. why not return the vault client and rename the method `createHashicorpVaultClient`?